src/pyuvdata/utils/phasing.py

```diff
@@ -1117,8 +1115,6 @@ def transform_icrs_to_app(
     if ra_coord.ndim == 0:
         ra_coord.shape += (1,)
         dec_coord.shape += (1,)
-        if pm_ra_coord is not None:
-            pm_ra
```

Comment:
I think this should actually be:

```
if pm_ra_coord is not None:
    pm_ra_coord.shape += (1,)
if pm_dec_coord is not None:
    pm_dec_coord.shape += (1,)
```

Not sure what happened there -- many something got lost in translation in the "Great Utils Migration of 2024"?

Reply:
Making that change results in a DeprecationWarning: 

```
tests/utils/test_phasing.py::test_astrometry_icrs_to_app[True]
  /Users/bryna/miniforge3/envs/rasg/lib/python3.12/site-packages/novas/compat/compat.py:3941: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    _make_cat_entry(ctypes.create_string_buffer(star_name.encode(), 51),
```

Reply:
If I run with `-W error` I get:

```
========================================================== FAILURES ==========================================================
_____________________________________________ test_astrometry_icrs_to_app[True] ______________________________________________
[gw6] darwin -- Python 3.12.4 /Users/bryna/miniforge3/envs/rasg/bin/python3.12

astrometry_args = {'app_coord': <SkyCoord (ICRS): (ra, dec) in deg
    [(141.72909621, 70.64308674), (141.72882194, 70.64297217),
     (...95246, 1.23295095, 1.23285455]), 'app_ra': array([2.47363937, 2.47363459, 2.47360784, 2.47347725]), 'dist': 73.31, ...}
use_extra = True

    @pytest.mark.filterwarnings('ignore:ERFA function "pmsafe" yielded 4 of')
    @pytest.mark.filterwarnings('ignore:ERFA function "utcut1" yielded 2 of')
    @pytest.mark.filterwarnings('ignore:ERFA function "d2dtf" yielded 1 of')
    @pytest.mark.parametrize("use_extra", [True, False])
    def test_astrometry_icrs_to_app(astrometry_args, use_extra):
        """
        Check for consistency beteen astrometry libraries when converting ICRS -> TOPP
    
        This test checks for consistency in apparent coordinate calculations using the
        three different libraries that are available to pyuvdata, namely: astropy, pyERFA,
        and python-novas. Between these three, we expect agreement within 100 µas in
        most instances, although for pyuvdata we tolerate differences of up to 1 mas since
        we don't expect to need astrometry better than this.
        """
        pytest.importorskip("novas")
        pytest.importorskip("novas_de405")
        # Do some basic cross-checking between the different astrometry libraries
        # to see if they all line up correctly.
        astrometry_list = ["novas", "erfa", "astropy"]
        coord_results = [None, None, None, None]
    
        # These values were indepedently calculated using erfa v1.7.2, which at the
        # time of coding agreed to < 1 mas with astropy v4.2.1 and novas 3.1.1.5. We
        # use those values here as a sort of history check to make sure that something
        # hasn't changed in the underlying astrometry libraries without being caught
        precalc_ra = np.array(
            [2.4736400623737507, 2.4736352750862760, 2.4736085367439893, 2.4734781687162820]
        )
        precalc_dec = np.array(
            [1.2329576409345270, 1.2329556410623417, 1.2329541289890513, 1.2328577308430242]
        )
    
        coord_results[3] = (precalc_ra, precalc_dec)
    
        kwargs = {}
        extra_args = ["pm_ra", "pm_dec", "vrad", "dist"]
        if use_extra:
            for key in extra_args:
                kwargs[key] = astrometry_args[key]
        else:
            # don't compare to precalc if not using extra arguments
            coord_results = coord_results[:-1]
    
        for idx, name in enumerate(astrometry_list):
>           coord_results[idx] = phs_utils.transform_icrs_to_app(
                time_array=astrometry_args["time_array"],
                ra=astrometry_args["icrs_ra"],
                dec=astrometry_args["icrs_dec"],
                telescope_loc=astrometry_args["telescope_loc"],
                epoch=astrometry_args["epoch"],
                astrometry_library=name,
                **kwargs,
            )

tests/utils/test_phasing.py:1551: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/pyuvdata/utils/phasing.py:1263: in transform_icrs_to_app
    cat_entry = novas.make_cat_entry(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

star_name = 'dummy_name', catalog = 'GKK', star_num = 156, ra = np.float64(9.427065589219145)
dec = np.float64(70.70299191914359), pm_ra = array([4.06472083]), pm_dec = array([45.6])
parallax = np.float64(13.640703860319192), rad_vel = np.float64(31.4)

    def make_cat_entry(star_name, catalog, star_num, ra, dec, pm_ra, pm_dec,
                       parallax, rad_vel):
        """
        Create an instance of CatEntry containing catalog data for a star or
        "star-like" object.
    
        Parameters
        ----------
        star_name : string
            Object name. Max length determined by SIZE_OF_OBJ_NAME
            defined in ``novas.h``; defaults to 50.
        catalog : string
            Catalog identifier (e.g. HIP = Hipparcos, TY2 = Tycho-2).
            Max length determined by SIZE_OF_CAT_NAME defined in
            ``novas.h``; defaults to 3.
        star_num : integer
            Object number in the catalog.
        ra : float
            Right ascension of the object (hours).
        dec : float
            Declination of the object (degrees).
        pm_ra : float (optional)
            Proper motion in right ascension (milliarcseconds/year).
        pm_dec : float (optional)
            Proper motion in declination (milliarcseconds/year).
        parallax : float (optional)
            Parallax (milliarcseconds).
        rad_vel : float (optional)
            Radial velocity (kilometers/second).
    
        Returns
        -------
        star : CatEntry
            Instance of CatEntry type object containing the input data.
    
        Notes
        -----
        .. [N1] This function is equivalent to calling the object with
            arguments,e.g., CatEntry(star_name, catalog, star_num, ...);
            this function exists for the purpose of direct compatibility
            with NOVAS C.
    
        """
    
        if not 0.0 <= ra < 24.0:
            raise ValueError(_hour_range_err.format(name='ra'))
        if not -90.0 <= dec <= 90.0:
            raise ValueError(_elev_range_err.format(name='dec'))
        if parallax < 0.0:
            raise ValueError(_neg_err.format(name='parallax'))
    
        _make_cat_entry = novaslib.make_cat_entry
        _make_cat_entry.argtypes = (ctypes.c_char*51, ctypes.c_char*4,
                                    ctypes.c_long, ctypes.c_double,
                                    ctypes.c_double, ctypes.c_double,
                                    ctypes.c_double, ctypes.c_double,
                                    ctypes.c_double, ctypes.POINTER(CatEntry))
        _make_cat_entry.restype = ctypes.c_short
        _make_cat_entry.errcheck = _check_c_errors
        _make_cat_entry.c_errors = {
            1: (ValueError, "from C function 'make_cat_entry': length of 'star_name' out of bounds."),
            2: (ValueError, "from C function 'make_cat_entry': length of 'catalog' out of bounds.")
            }
    
        star = CatEntry()
    
>       _make_cat_entry(ctypes.create_string_buffer(star_name.encode(), 51),
                        ctypes.create_string_buffer(catalog.encode(), 4), star_num,
                        ra, dec, pm_ra, pm_dec, parallax, rad_vel,
                        ctypes.byref(star))
E       ctypes.ArgumentError: argument 6: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)

../../../miniforge3/envs/rasg/lib/python3.12/site-packages/novas/compat/compat.py:3941: ArgumentError
```

Reply:
The same line appears in v2.4.5, so I don't think it was introduced in the utility migration:

https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/86420859041e341dcaaf52eaca77f3f6f17dea60/pyuvdata/utils.py#L2739

Reply:
Since that line was clearly not doing anything before and the tests were passing without warnings or errors, I removed it. But happy for you to take a closer look and see if there's something else that should be happening.

Reply:
Okay, I found the issue and fixed it in 5676bfe -- the above check was added back, as was a tiny bit of cleanup in the call to the NOVAS version of the astrometry calculation.